### ISSUE TITLE
No Issue - How to donate a font?

### ISSUE THREAD

**Issue by bushy555** [robhagemans/hoard-of-bitfonts#19]
Hi 

(Unsure how to contact you directly coz Im too dumb to work the github gui.  So apologies on this)

Just wanting to know how I can donate the font to Laser200 / Laser310 / VZ200 / VZ300 8-bit computer font?
Read your article on hack-a-day website on 5th Feb.

Cheers,
Dave

**Comment by robhagemans** [comment]
Hi Dave, you can clone the repository to your local system and create a pull request - this way I can merge and  you'll show 
up as a contributor on github. Alternatively you can send them to me or attach to a post here and I can add them, but git won't add your name to the commit. Either way works for me...

**Comment by bushy555** [comment]
Hi

I'm too dumb to work out Pull requests. In fact Im so dumb that I can't even get into any my four dropbox accounts to share this. I gave up.  Zipped and is here to download : https://easyupload.io/1y7cgs

Included : Binary extraction (768 bytes), Font files (3072 bytes), 
Font size : 7x12 (from memory)

Computer : VZ200  and  VZ300.  Also known as Laser210 and Laser310.
Company : Video Technology  Hong Kong
Manufacturing era: 1983-1985.
CPU: Z80A @ 3.5Mhz
Video chip: Motorola 6847.

**Comment by robhagemans** [comment]
Hi, thanks again - the archive seems to contain these files:

      -rw-rw-r-- 1 rob rob   626 Jan  3  2021 screnLow.asm
      -rw-rw-r-- 1 rob rob  2637 Jan  3  2021 VZgame.asm
      -rw-rw-r-- 1 rob rob 13568 Jan  3  2021 vzhappy.asm
      -rw-rw-r-- 1 rob rob  4172 Jan  3  2021 VZHAPPY.OBJ
      -rw-rw-r-- 1 rob rob  7222 Jan  3  2021 vzScreen.asm
      -rw-rw-r-- 1 rob rob  2124 Jan  3  2021 VZSCREEN.OBJ

I don't actually see anything that looks like a font, the `.OBJ` files contain two bitmaps of a somewhat crudely drawn face but that is all. I'm assuming the `.asm` files encode these though I haven't decoded them. Please let me know what you're expecting to find....

**Comment by bushy555** [comment]
Hi

Oops. Im a twit.  Please ignore that file.       "Don't mind me - Im a goose!"
I somehow, very successfully, managed to upload the wrong zip file. Apologies for my dumbness.

Second attempt :     https://easyupload.io/dv8w5w
I downloaded it to ensure that it had the correct files, which it does.

**Comment by robhagemans** [comment]
Thanks! I now have these files:        

        -rw-rw-r-- 1 rob rob 3.0K Nov 12  2019 'VZ200_Character_Set_(1983)_v1.bin'
        -rw-rw-r-- 1 rob rob  768 Sep 12  2004  VZ200_chr.bin
        -rw-rw-r-- 1 rob rob 3.0K Oct  8 21:14  VZ200_default.fnt
        -rw-rw-r-- 1 rob rob  16K Oct  8 21:14  VZ200_default_font.aseprite
        -rw-rw-r-- 1 rob rob  768 Sep 12  2004  VZ200_mess_ntscround_chr.bin
        -rw-rw-r-- 1 rob rob  768 Sep 12  2004  VZ200_mess_ntscsquare_chr.bin
        -rw-rw-r-- 1 rob rob  768 Apr 17  2022  VZ200_outline.fnt
        -rw-rw-r-- 1 rob rob 3.0K Oct  9 12:18  VZ200_spaceinvader.fnt

The `.bin` and `.fnt` files are all 12-pixel high raw binary files so they're easy to extract. I was wondering, do you have any background to where the different fonts come from?  E.g. different versions of the VZ200? I imagine the Space Invader font belongs to the game of that name, but there seem to be various different 'default' fonts as well

**Comment by SlashNetUA** [comment]
Hi!

I've tried to use git at first time to add some files, but I failed :,(

I've run next commands:
gh auth login
>gh repo clone robhagemans/hoard-of-bitfonts
>git add elan_enterprise/*
>git commit -m "Added Elan Enterprise official and user fonts"
>git add videoton_tvc/*
>git commit -m "Added Videoton TVC official fonts"
All ok
but when I run "git push" I have error:
remote: Permission to robhagemans/hoard-of-bitfonts.git denied to SlashNetUA.
fatal: unable to access 'https://github.com/robhagemans/hoard-of-bitfonts.git/': The requested URL returned error: 403

Where could I make a mistake?

**Comment by robhagemans** [comment]
> All ok but when I run "git push" I have error: **remote: Permission to robhagemans/hoard-of-bitfonts.git denied to SlashNetUA. fatal: unable to access 'https://github.com/robhagemans/hoard-of-bitfonts.git/': The requested URL returned error: 403**
> 
> Where could I make a mistake?

Looks like you're trying to  push directly to this repo - which you can't as it is my personal repo and you don't have write access. 

You need to create a pull request; easiest is probably to fork the repo through Github, you can then push to your own fork and create a pull request through the Github interface. 

It may also be possible directly through the command line, please check Github documentation.

**Comment by MarkJeronimus** [comment]
I like this "thread". It's a nice method for people to submit fonts without having to go through monobit and git pull requests.

Just leaving this here: https://we.tl/t-u683YUmxwt (93 8x16 VGA fonts of various code pages)

**Comment by robhagemans** [comment]
@bushy555 - an update on the VZ200. From what I read this used an MC6847 character generator which was also used in the TRS-80 CoCo (even though the VZ200 was otherwise more like the TRS-80 Model 1). 

Indeed as far as I can see the character set in `VZ200_Character_Set_(1983)_v1.bin` is identical to `mc6847t1.yaff` included there. I've recently cleaned that folder up as the MAME character ROMs turned out to be not authentic (this has since been fixed).  

So I don't think we need to include this separately .
